PWGJE/Tasks/hffragmentationfunction.cxx

```diff
@@ -43,6 +46,19 @@ using namespace o2;
 using namespace o2::framework;
 using namespace o2::framework::expressions;
 
+// calculate delta phi such that 0 < delta phi < 2*pi
+double DeltaPhi(double phi1, double phi2)
+{
+  // Compute the absolute difference between phi1 and phi2
+  double dphi = std::abs(phi1 - phi2);
+  if (dphi > M_PI) {
+    // subtract 2pi if the difference if bigger than pi
+    dphi = dphi - 2 * M_PI;
+  }
+
+  return dphi;
+}
```

Comment:
There a several issues with this.
- For calculating a distance, you need -Pi < deltaPhi <= +Pi.
- Use `RecoDecay::constrainAngle` to adjust the range properly: https://github.com/AliceO2Group/O2Physics/blob/4ecf524826c7279cb172e35ac2b390f9696dd84c/Common/Core/RecoDecay.h#L214-L220
- `M_PI` should not be used. Use `o2::constants::math::PI`.
- Names of functions start with lowercase letters.

Reply:
Thanks for the explanation Vit, I included your suggestions.
And as for the deltaPhi() function, I believe that as I am interested only in the absolute value of the angle difference, the function was already correct as it was. However I modified and included the `RecoDecay::constrainAngle` function.